src/testing/slim_check/tactics.lean

```diff
@@ -0,0 +1,96 @@
+
+import testing.slim_check.testable
+
+namespace tactic
+open slim_check
+
+private meta def applye (e : pexpr) : tactic unit := do
+() <$ (to_expr e >>= tactic.apply)
+
+/-- build an instance of testable for the given proposition
+  -/
+meta def is_testable : tactic unit := do
+(do
+tactic.target >>= instantiate_mvars >>= tactic.change,
+`(testable %%e) ← target,
+match e with
+ | (expr.pi n bi d t) :=
+   if bi = binder_info.inst_implicit then do
+      h ← tactic.assert `h d,
+      solve1 apply_instance,
+      applye ``(@slim_check.test_one _ _ %%h _),
+      is_testable
+   else do
+    p ← is_prop d,
+    let var := reflect $ to_string n,
+    let mk_testable_inst := (do
+          h ← to_expr ``(testable %%t) >>= λ e, tactic.assert `h (expr.pi n bi d e),
+          solve1 (tactic.intro1 >> is_testable)),
+    if p then do
+       mk_testable_inst,
+       tactic.applyc `slim_check.imp_dec_testable
+    else if d = `(Type) then do
+      let t' := expr.instantiate_local n `(ℤ) t,
+      h ← to_expr ``(testable %%t') >>= tactic.assert `h,
+      solve1 is_testable,
+      applye ``(slim_check.test_one _ _ ℤ (some (%%var,"ℤ"))) ; apply_instance
+      -- let specialize := (λ (sp : expr) (nm : string), do
+      --     let t' := expr.instantiate_local n sp t,
+      --     h ← to_expr ``(testable %%t') >>= tactic.assert `h,
+      --     solve1 is_testable,
+      --     let type := reflect nm,
+      --     applye ``(slim_check.test_one _ _ %%sp (some (%%var,%%type)))
+      --       ; apply_instance),
+      -- apply ``(slim_check.combine_testable _ [_,_] _),
+      -- specialize `(ℤ) "ℤ",
+      -- specialize `(list ℤ) "list ℤ",
+      -- apply ``(nat.zero_lt_succ)
```

Comment:
Dead code?

Reply:
I'm thinking of reviving it. It's a generalization that is still under development